test/helper/e2e/actions/cloud/provider.go

```diff
@@ -276,6 +279,23 @@ func (a *ProviderAction) SetupPrivateEndpoint(request PrivateEndpointRequest) *P
 	return nil
 }
 
+func (a *ProviderAction) RetryGCPCreateEndpoint(maxRetries int, name, region, subnet, target string, index int) (string, string, error) {
```

Comment:
this is too naive, please use exponential backoff retries, see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/wait/backoff.go#L477

Reply:
Will do. But please note this is test helper code, quite far from production requirements.

Reply:
This is not about production requirements.

Remote APIs (in this case GCP) cannot be retried in a brute-force manner in endless loops. You will most likely hit a rate-limiter on the GCP side.

Hence, retryring things on remote APIs needs to have:
- jitter
- delay between retries
- exponential backoff between retries

This is especially important for test code.